src/workflows/recipes/components/FilterObjectForm.js

```diff
@@ -0,0 +1,803 @@
+/* eslint-disable react/jsx-boolean-value */
+import {
+  AutoComplete,
+  Button,
+  Checkbox,
+  Col,
+  InputNumber,
+  Radio,
+  Row,
+  Select,
+  Tabs,
+  Tag,
+} from 'antd';
+import { fromJS, List, Map } from 'immutable';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import QueryRecipeFilters from 'console/components/data/QueryRecipeFilters';
+import FormItem from 'console/components/forms/FormItem';
+
+// XXX Why is this needed?!?
+const TabsPane = Tabs.TabPane;
+const CheckboxGroup = Checkbox.Group;
+const RadioGroup = Radio.Group;
+const Option = Select.Option;
+
+// Every filter object is an Map (in an List) that looks something like this:
+//  {"type": "channel", "channels": ["beta"]}
+// We need a mapping from "channel" to "channels", for each, to know what to
+// call it when we serialize this list into a Map where entries look like this:
+//  {"channels": ["beta"]}
+// It is not safe to assume that the "type" can just be the plural form of the word.
+export const MAPPING = {
+  channel: 'channels',
+  country: 'countries',
+  locale: 'locales',
+  version: 'versions',
+};
+
+// In the UI we present the labels but in the data structure we store it by its value.
+// This mapping needs to be ordered to generate the <Select><Option/><Option/>... components.
+// Also, this is used to *display* the human readable version of each type when viewing
+// the details of the filter object.
+export const SAMPLING_TYPES = [
+  { value: 'bucketSample', label: 'Bucket' },
+  { value: 'stableSample', label: 'Stable' },
+  { value: '', label: 'None' },
+];
+
+/* Return a Map from a List.
+  Each item in the list is expected to be something like this:
+
+      [{"type": "age", "ages": [1,2,3]}, ]
+
+  Which is then transformed to this:
+
+      {"ages": [1,2,3]}
+
+  The list of valid keys needs to be known in advance. */
+export const serializeFilterObjectToMap = list => {
```

Comment:
Why an arrow function instead of `export function serializeFilterObjectToMap(list) {`?

Reply:
I don't know. I see that sometimes people write it as a function and sometimes this format. I never know which "best".